### PR TITLE
Chord diagram

### PR DESCRIPTION
**Pull request by IlincaTudose**

### CODE REVIEW

File: web/src/main/webapp/WEB-INF/views/chordDiagram.jsp [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+<t:genericpage>
+
+    <jsp:attribute name="title">Chord diagram</jsp:attribute>
+
+    <jsp:attribute name="bodyTag">
+		<body class="gene-node no-sidebars small-header">
+	</jsp:attribute>
+
+
```
Comment (jmason-ebi) [inline web/src/main/webapp/WEB-INF/views/chordDiagram.jsp]: Also override the breadcrumb attribute.

File: data-model-solr/src/main/java/org/mousephenotype/cda/solr/service/PostQcService.java [hunk 2]
```diff
@@ -121,6 +128,141 @@ public void postSetup() {
...
     }
 
+    /**
+     * Used by
+     * @param topLevelMpTerms the mp terms are used with AND. If not null returns data for genes that have ALL phenotypes in the passed list.
+     * @return
+     */
+    public JSONObject getPleiotropyMatrix(List<String> topLevelMpTerms) throws IOException, SolrServerException {
+
+        TreeMap<String, TreeMap<String, Integer>> matrix = new TreeMap<>();
```
Comment (jmason-ebi) [inline data-model-solr/src/main/java/org/mousephenotype/cda/solr/service/PostQcService.java]: use generic Map

Comment (jmason-ebi) [inline data-model-solr/src/main/java/org/mousephenotype/cda/solr/service/PostQcService.java]: Discussed, it's good to use treemap

File: web/src/main/webapp/WEB-INF/views/chordDiagram.jsp [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+    <jsp:attribute name="header">
+
+    <script src="https://d3js.org/d3.v4.min.js"></script>
```
Comment (jmason-ebi) [inline web/src/main/webapp/WEB-INF/views/chordDiagram.jsp]: Use protocol relative url?

File: data-model-solr/src/main/java/org/mousephenotype/cda/solr/service/PostQcService.java [hunk 4]
```diff
@@ -121,6 +128,141 @@ public void postSetup() {
...
+        result.put("matrix", jsonMatrix);
+        result.put("labels", labelList);
+
+        return result;
+
+    }
+
+    public String getPleiotropyDownload(List<String> topLevelMpTerms) throws IOException, SolrServerException {
+
+        TreeMap<String, TreeMap<String, Integer>> matrix = new TreeMap<>();
```
Comment (IlincaTudose) [inline data-model-solr/src/main/java/org/mousephenotype/cda/solr/service/PostQcService.java]: I don't know about "needed" but I changed it to a matrix. Still think it's harder to follow a bidimensional array and it's parallel label array than just a map.

File: web/src/main/webapp/WEB-INF/views/chordDiagram.jsp (hunk 1 above)
Comment (IlincaTudose) [inline web/src/main/webapp/WEB-INF/views/chordDiagram.jsp]: done

**Comment by IlincaTudose** [comment]
Changes done

File: data-model-solr/src/main/java/org/mousephenotype/cda/solr/service/PostQcService.java (hunk 4 above)
Comment (IlincaTudose) [inline data-model-solr/src/main/java/org/mousephenotype/cda/solr/service/PostQcService.java]: Heh, wrong method. Removed that now. I started the matrix refactoring on the other method after our off-line discussion.

**Pull request merged**